src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -610,6 +612,12 @@ def create_vm(cmd, vm_name, resource_group_name, image=None, size='Standard_DS1_
     if secrets:
         secrets = _merge_secrets([validate_file_or_dict(secret) for secret in secrets])
 
+    if key_encryption_key:
+        key_encryption_keyvault = key_encryption_keyvault
+        if '://' not in key_encryption_key:  # appears a key name
+            key_encryption_key = get_keyvault_key_url(
+                cmd.cli_ctx, (parse_resource_id(key_encryption_keyvault))['name'], key_encryption_key)
```

Comment:
please move this to the `_validators.py`

Reply:
Done